docs/metrics.md

```diff
@@ -0,0 +1,587 @@
+   is only applicable to Delta temporality)
+
+Therefore, the actual cardinality in your metrics backend can be orders of
+magnitude higher than what any single OTel SDK process handles in an export
+cycle.
+
+#### Cardinality Limits - Implications
+
+Cardinality limits are enforced for each export interval, meaning the metrics
+aggregation system only allows up to the configured cardinality limit of unique
+attribute combinations per metric. Understanding how this works in practice is
+important:
+
+* **Cardinality Capping**: When the limit is reached within an export interval,
+  any new attribute combination is not individually tracked but instead folded
+  into a single aggregation with the attribute `{"otel.metric.overflow": true}`.
+  This preserves the overall accuracy of aggregates (such as Sum, Count, etc.)
+  even though information about specific attribute combinations is lost. Every
+  measurement is accounted for - either with its original attributes or within
+  the overflow bucket.
+
+* **Temporality Effects**: The impact of cardinality limits differs based on the
+  temporality mode:
+  
+  * **Delta Temporality**: The SDK "forgets" the state after each
+    collection/export cycle. This means in each new interval, the SDK can track
+    up to the cardinality limit of completely different attribute combinations.
+    Over time, your metrics backend might see far more than the configured limit
+    of unique combinations from a single process.
+  
+  * **Cumulative Temporality**: Since the SDK maintains state across export
+    intervals, once the cardinality limit is reached, new attribute combinations
+    will continue to be folded into the overflow bucket. The total number of
+    distinct attribute combinations exported cannot exceed the cardinality limit
+    for the lifetime of that metric instrument.
+
+* **Impact on Monitoring**: While cardinality limits protect your system from
+  unbounded resource consumption, they do mean that high-cardinality dimensions
+  may not be fully represented in your metrics. Since cardinality capping can
+  cause metrics to be folded into the overflow bucket, it becomes impossible to
+  predict which specific attribute combinations were affected across multiple
+  collection cycles or different service instances.
+
+  This unpredictability creates several important considerations when querying
+  metrics in any backend system:
+  
+  * **Total Accuracy**: OpenTelemetry Metrics always ensures the total
+    aggregation (sum of metric values across all dimensions) remains accurate,
+    even when overflow occurs.
+  
+  * **Attribute-Based Query Limitations**: Any metric query based on specific
+    attributes could be misleading, as it's possible those dimensions were
+    folded into the overflow bucket due to cardinality capping.
+  
+  * **All Attributes Affected**: When overflow occurs, it's not just
+    high-cardinality attributes that are affected. The entire attribute
```

Comment:
This description is not so different from the above one. I think we can rearrange these sections for simpler understanding:

This unpredictability creates several important considerations when querying metrics in any backend system:
* **Total Accuracy**: ...
* **Attributes-based querying**
  * **Only partial information retained:** (this would be the "How many red apples were sold?" example). Measurements with a superset of dimensions could be folded into overflow. We only retained information for Downtown based measurements here.
  * **No information retained:** This would be the "How many items were sold in Midtown?" example. All measurements related to Midtown were folded into overflow.
  
 Does that make sense?